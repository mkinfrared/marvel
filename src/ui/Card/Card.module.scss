@import "styles/colors";

.Card {
  border: 1px solid $secondary-main;

  .image {
    height: 300px;
    margin-bottom: 10px;
    width: 100%;

    img,
    svg {
      height: 100%;
      object-fit: contain;
      width: 100%;
    }
  }

  > p {
    margin-bottom: 10px;
  }
}
