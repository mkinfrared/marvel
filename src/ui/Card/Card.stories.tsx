import { Meta, Story } from "@storybook/react/types-6-0";
import React from "react";

import { Card } from "./Card";
import { CardProps } from "./Card.type";

export default {
  title: "UI/Card",
  component: Card
} as Meta;

const Template: Story<CardProps> = (args) => <Card {...args}>Hello</Card>;
const Primary = Template.bind({});

Primary.args = {
  imageSrc: "http://i.annihil.us/u/prod/marvel/i/mg/6/f0/5239b5e7d7f70.jpg",
  title: "Dr. Strange"
};

export { Primary };
