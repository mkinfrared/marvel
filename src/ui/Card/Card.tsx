import React from "react";

import Image from "ui/Image";

import css from "./Card.module.scss";
import { CardProps } from "./Card.type";

const Card = ({ imageSrc, title }: CardProps) => {
  return (
    <div className={css.Card}>
      <div className={css.image}>
        <Image src={imageSrc} loadHeight={300} loadWidth={200} />
      </div>
      <p>{title}</p>
    </div>
  );
};

export { Card };

export default React.memo(Card);
