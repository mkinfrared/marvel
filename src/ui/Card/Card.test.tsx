import { render } from "@testing-library/react";
import React from "react";

import Card from "./Card";

jest.mock("ui/Image", () => () => "<Image />");

describe("<Card />", () => {
  const Component = Card;
  const title = "marklar";
  const imageSrc = "foobar";
  const props = { title, imageSrc };

  it("should be defined", () => {
    const { container } = render(<Component {...props} />);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });
});
