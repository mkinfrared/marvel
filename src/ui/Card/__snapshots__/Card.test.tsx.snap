// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> match the snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <div
      class="image"
    >
      &lt;Image /&gt;
    </div>
    <p>
      marklar
    </p>
  </div>
</div>
`;
